### PR TITLE
Add config to get the totalResults by user count

### PR DESCRIPTION
**Pull request by asha15**
### Proposed changes in this pull request

Resolve the issue: https://github.com/wso2/product-is/issues/12813

When the SCIM API is invoked with a filter, the current behavior involves returning a predefined value for the totalResults parameter based on the MaxUserNameListLength, rather than reflecting the actual results that match the filter criteria. This fix addresses the issue by accurately determining the user count that corresponds to the given filter. To enable this fix, the following config should be added

```
[scim2]
enable_group_based_user_filter_improvements = true
```

Related PRs: https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/533, https://github.com/wso2/carbon-kernel/pull/3906

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 1]
```diff
@@ -1414,6 +1414,7 @@
              Supported versions : IS 5.10.0 onwards
         -->
         <ConsiderTotalRecordsForTotalResultOfLDAP>{{scim2.consider_total_records_for_total_results_of_ldap}}</ConsiderTotalRecordsForTotalResultOfLDAP>
+        <EnableRetrieveTotalResultsByUserCount>{{scim2.enable_retrieve_totalresults_By_user_count}}</EnableRetrieveTotalResultsByUserCount>
```
Comment (bhagyasakalanka) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:1417]: Lets add this to default.json file as well

Comment (dhaura) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:1417]: Rename config: `EnableGroupBasedUserFiltering`

Comment (dhaura) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:1417]: Add improvement details in a comment

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 2]
```diff
@@ -976,6 +976,17 @@
              Supported versions : IS 7.0.0 onwards
         -->
         <UserEndpointMaxItemsPerPageEnabled>true</UserEndpointMaxItemsPerPageEnabled>
+
+        <!--
+            This configuration enhances the behavior of the SCIM API when invoked with a group-based filter.
```
Comment (dhaura) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:981]:
```suggestion
            This configuration enhances the behavior of the SCIM2 API when invoked with a group-based filter.
```

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 3]
```diff
@@ -976,6 +976,17 @@
...
+            It determines the totalResults based on the total user count corresponding to the given group-based filter.
+            Additionally, when invoking the user endpoint with a groups filter by passing a role (hybrid or system roles),
+            it returns an empty list when role group separation is enabled, instead of returning users linked to the passed role.
+            Default value : false
+            Supported versions : IS 7.0.0 onwards
+        -->
+        <!--<EnableGroupBasedUserFilteringImprovements>false</EnableGroupBasedUserFilteringImprovements>-->
```
Comment (mpmadhavig) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:988]: What's the reason for commenting this out?

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 4]
```diff
@@ -976,6 +976,17 @@
...
         -->
         <UserEndpointMaxItemsPerPageEnabled>true</UserEndpointMaxItemsPerPageEnabled>
+
+        <!--
+            This configuration enhances the behavior of the SCIM API when invoked with a group-based filter.
+            It determines the totalResults based on the total user count corresponding to the given group-based filter.
+            Additionally, when invoking the user endpoint with a groups filter by passing a role (hybrid or system roles),
+            it returns an empty list when role group separation is enabled, instead of returning users linked to the passed role.
+            Default value : false
+            Supported versions : IS 7.0.0 onwards
```
Comment (dhaura) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:986]: Will we be back porting this fix as a patch to IS 7.0.0?

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml (hunk 2 above)
Comment (asha15) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml]: Fixed with : https://github.com/wso2/carbon-identity-framework/pull/5533/commits/9e8d831026a3b754350a555c963d1cb383a112d0

**Review by sumedhe — APPROVED** [review]

**Pull request merged**